parsers/manifest_parser.go

```diff
@@ -262,6 +257,15 @@ func (dm *YAMLParser) ComposeActions(mani *ManifestYAML, manipath string) (ar []
 				}
 
 				wskaction.Exec.Kind = kind
+
+				action.Location = filePath
+				dat, err := utils.Read(filePath)
+				utils.Check(err)
+				code := string(dat)
+                		if ext == ".zip" || ext == ".jar" {
```

Comment:
I tested JAR with HelloWorld application from [here](https://console.ng.bluemix.net/docs/openwhisk/openwhisk_actions.html#openwhisk_actions_java). We do not have support for this application. wskdeploy fails with:

```
Response body received:
{
  "error": "The request content was malformed:\n'main' must be a string defined in 'exec' for 'java' actions",
  "code": 4001042
}
``` 

For actions in Java runtime, wsk CLI supports "--main" param which (I assume) might not be supported in wskdeploy. While creating an action from the jar file where main is inside of a class, we have to specify the class name which contains main function.

```
wsk action create helloJava hello.jar --main Hello
``` 

I am using this manifest file:

```
package:
    name: helloworldjar
    actions:
        helloworld:
            location: src/hello.jar
            runtime: java
            main: Hello
```

I am creating separate issue to address JAR failure.